diff --git a/faiss/impl/mapped_io.cpp b/faiss/impl/mapped_io.cpp
index 32486a9e6..58bfe44c8 100644
--- a/faiss/impl/mapped_io.cpp
+++ b/faiss/impl/mapped_io.cpp
@@ -171,12 +171,12 @@ struct MmappedFileMappingOwner::PImpl {
         const int fd = _fileno(f);
         if (fd == -1) {
             // no good
-            FAISS_THROW_FMT("could not get a HANDLE");
+            FAISS_THROW_MSG("could not get a HANDLE");
         }
 
         HANDLE file_handle = (HANDLE)_get_osfhandle(fd);
         if (file_handle == INVALID_HANDLE_VALUE) {
-            FAISS_THROW_FMT("could not get an OS HANDLE");
+            FAISS_THROW_MSG("could not get an OS HANDLE");
         }
 
         // get the size of the file
